src/hooks/useCopy.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+
+const delay = 1_000;
+
+/**
+ * A React hook that provides copy-to-clipboard functionality with temporary visual feedback.
+ *
+ * Returns a copy handler and a boolean indicating whether the data was recently copied.
+ * The copied state automatically resets after 1 second.
+ */
+function useCopy(data: string): [() => void, boolean] {
```

Comment:
I wonder if this should accept an optional `event?: SyntheticEvent<Element>`, where the function can call `event?.preventDefault()`. It seems like a pretty common use case.